crates/red_knot_python_semantic/resources/mdtest/unreachable.md

```diff
@@ -218,26 +255,24 @@ def f():
 
 In the example below, since we use `x` in the `inner` function, we use the "public" type of `x`,
 which currently refers to the end-of-scope type of `x`. Since the end of the `outer` scope is
-unreachable, we treat `x` as if it was not defined. This behavior can certainly be improved.
+unreachable, we need to make sure that we do not emit an `unresolved-reference` diagnostic:
 
 ```py
 def outer():
     x = 1
 
     def inner():
-        return x  # Name `x` used when not defined
+        reveal_type(x)  # revealed: Unknown
```

Comment:
The fact that we reveal `Unknown` here is tracked in https://github.com/astral-sh/ruff/issues/15777